drivers/temperature/adt7420/adt7420.c

```diff
@@ -40,55 +40,107 @@
 /******************************************************************************/
 /***************************** Include Files **********************************/
 /******************************************************************************/
+#include <stdint.h>
 #include <stdlib.h>
+#include "spi.h"
+#include "i2c.h"
+#include "error.h"
 #include "adt7420.h"
 
+const struct adt7420_chip_info chip_info[] = {
+	[ID_ADT7410] = {
+		.resolution = 16,
+		.communication = I2C,
+	},
+	[ID_ADT7420] = {
+		.resolution = 16,
+		.communication = I2C,
+	},
+	[ID_ADT7422] = {
+		.resolution = 16,
+		.communication = I2C,
+	},
+	[ID_ADT7310] = {
+		.resolution = 16,
+		.communication = SPI,
+	},
+	[ID_ADT7320] = {
+		.resolution = 16,
```

Comment:
is resolution needed if is the same for all?
Can it go as before in adt7420_dev ?

Reply:
I see many devices from the family that don't have 16-bit resolution so it's best to keep resolution in each entry:
https://eepower.com/new-industry-products/analog-devices-intros-adt7xxx-converter-family/#

Reply:
Resolution field remains unaltered, considering devices that do not have a 16-bit resolution